quil-rs/src/expression/mod.rs

```diff
@@ -145,20 +207,27 @@ impl InfixExpression {
 /// Note that when comparing Quil expressions, any embedded NaNs are treated as *equal* to other
 /// NaNs, not unequal, in contravention of the IEEE 754 spec.
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
+#[pyclass(module = "quil.expression", eq, frozen, hash, subclass)]
 pub struct PrefixExpression {
     pub operator: PrefixOperator,
     pub expression: ArcIntern<Expression>,
 }
 
+#[pymethods]
 impl PrefixExpression {
-    pub fn new(operator: PrefixOperator, expression: ArcIntern<Expression>) -> Self {
+    #[new]
+    pub fn new(
+        operator: PrefixOperator,
+        #[pyo3(from_py_with = intern_py)] expression: ArcIntern<Expression>,
+    ) -> Self {
         Self {
             operator,
             expression,
         }
     }
 }
 
+// TODO: PartialEq/Eq is inconsistent with Hash: https://github.com/rigetti/quil-rs/issues/458
```

Comment:
😬

Reply:
Yeah, the good news is, it _probably_ doesn't come up very often.

Reply:
```suggestion
// TODO (#458): PartialEq/Eq is inconsistent with Hash
```

I think this tends to be how we record things like this in code – Kalan once advised me to avoid unannotated TODOs